name.abuchen.portfolio/src/name/abuchen/portfolio/datatransfer/pdf/OnvistaPDFExtractor.java

```diff
@@ -499,11 +499,27 @@ private void addDividendeTransaction()
 
                 .oneOf(
                             // STK 50,000 21.04.2016 21.04.2016 EUR 0,200000
+                            
+                            section -> section
+                                    .attributes("date","exDate")
+                                    .match("^STK [\\.,\\d]+ (?<exDate>[\\d]{2}\\.[\\d]{2}\\.[\\d]{4})? (?<date>[\\d]{2}\\.[\\d]{2}\\.[\\d]{4}) [\\w]{3} [\\.,\\d]+$")
+                                    .assign((t, v) -> {
+                                        t.setDateTime(asDate(v.get("date")));
+                                        t.setExDateTime(asDate(v.get("exDate")));
+                                    })
+                            ,
                             // STK 1,000 17.11.2010 EUR 1,548250
                             section -> section
-                                    .attributes("date")
-                                    .match("^STK [\\.,\\d]+( [\\d]{2}\\.[\\d]{2}\\.[\\d]{4})? (?<date>[\\d]{2}\\.[\\d]{2}\\.[\\d]{4}) [\\w]{3} [\\.,\\d]+$")
-                                    .assign((t, v) -> t.setDateTime(asDate(v.get("date"))))
+                            .attributes("date")
+                            .match("^STK [\\.,\\d]+ (?<date>[\\d]{2}\\.[\\d]{2}\\.[\\d]{4}) [\\w]{3} [\\.,\\d]+$")
+                            .assign((t, v) -> t.setDateTime(asDate(v.get("date"))))
+                            ,
+                            
+                            //STK 33,000 06.06.2021
+                            section -> section
+                                       .attributes("date")
+                                       .match("^STK [\\.,\\d]+ (?<date>[\\d]{2}\\.[\\d]{2}\\.[\\d]{4}) $")
```

Comment:
`^STK [\\.,\\d]+ (?<date>[\\d]{2}\\.[\\d]{2}\\.[\\d]{4}) $`
Is the whitespace before the `$` really needed?

Reply:
not needed -> changed